engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -4590,10 +4650,20 @@ struct soul_cleave_t : public demon_hunter_attack_t
             ->debuffs.frailty->trigger(
                 timespan_t::from_seconds( p()->talent.vengeance.soulcrush->effectN( 2 ).base_value() ) );
       }
+
+      if ( t29_proc )
+      {
+        td( s->target )->debuffs.t29_vengeance_4pc->trigger();
+      }
     }
 
     void execute() override
     {
+      if ( p()->set_bonuses.t29_vengeance_4pc->ok() )
+      {
+        t29_proc = rng().roll( p()->set_bonuses.t29_vengeance_4pc->effectN( 2 ).percent() );
```

Comment:
This mechanic is a bit strange and while this may work it may not work for all methods of triggering (such as scheduled executes.)

It may be more correct to override demon_hunter_attack_t::snapshot_internal() and then add to the 
```
if ( p()->set_bonuses.t29_vengeance_4pc->ok() && flags & STATE_MUL_DA 
     && rng().roll( p()->set_bonuses.t29_vengeance_4pc->effectN( 2 ).percent() ) )
    state->da_multiplier *= p()->set_bonuses.t29_vengeance_4pc->effectN( 1 ).percent();
```

Reply:
I'll give that a shot!

Reply:
Should the trigger of the debuff go in the snapshot as well?

Reply:
No, that would be before there are any targets. I guess that's tricky... could make a custom action state perhaps. Seems overkill but may actually be the cleanest way. I'm not totally sure. Let me look at a few options.

Reply:
Since there are two spells affected by it, should they both use the same custom action state type?